src/Features/Core/Portable/StackTraceExplorer/StackFrameParserHelpers.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    internal class StackFrameParserHelpers
+    {
+        /// <summary>
+        /// Makes sure that the string at least somewhat resembles the correct form.
+        /// Does not check validity on class or method identifiers
+        /// Example line:
+        /// at ConsoleApp4.MyClass.ThrowAtOne(p1, p2,) 
+        ///   |-------------------||--------||-------| 
+        ///           Class          Method    Args   
+        /// </summary>
+        /// <remarks>
+        /// See https://docs.microsoft.com/en-us/dotnet/api/system.environment.stacktrace for more information
+        /// on expected stacktrace form
+        /// </remarks>
+
+        public static bool TryParseMethodSignature(string line, int start, int end, out TextSpan classSpan, out TextSpan methodSpan, out TextSpan argsSpan)
+        {
+            classSpan = default;
+            methodSpan = default;
+            argsSpan = default;
+
+            line = line[start..end];
+
+            var regex = new Regex(@"(?<class>([a-zA-Z0-9_<\s,>]+\.)+)(?<method>[a-zA-Z0-9_<\s,>]+)\((?<args>.*)\).*");
```

Comment:
this feels like it woudl not handle unicode.  I would expect us to instead use our helpers that identify legal identifier characters to handle this.

Reply:
this part is problematic: `(?<args>.*)\).*` as it can consume into the trailing section (which could def have a `)` in it.

Reply:
We'll do a full lexer/parser pass as discussed in a different PR